### ISSUE TITLE
t.p.b.StatefulStringProtocol has undocumented .state instance attribute

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5918]
|<img alt="devinj's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| devinj reported|
|-|-|
|Trac ID|trac#5918|
|Type|enhancement|
|Created|2012-08-25 03:32:30Z|
|Branch|https://github.com/twisted/twisted/tree/state-doc-5918|

http://twistedmatrix.com/trac/browser/tags/releases/twisted-12.1.0/twisted/protocols/basic.py#L846

http://twistedmatrix.com/documents/current/api/twisted.protocols.basic.StatefulStringProtocol.html

I'm not sure that instance attributes are universally supposed to be documented, but the attribute name "state" is useful enough for others (and nonobvious enough in subclasses, e.g. AMP) that maybe it should be documented. I ran into this problem last year when I used state as an instance attribute for a subclass, and found it very confusing. (At the time, I forgot to report the missing documentation, sorry.)

<details><summary>Searchable metadata</summary>

```
trac-id__5918 5918
type__enhancement enhancement
reporter__devinj devinj
priority__normal normal
milestone__ 
branch__branches_state_doc_5918 branches/state-doc-5918
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1345865550000000 1345865550000000
changetime__1361798892000000 1361798892000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [37212]) Branching to 'state-doc-5918'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [37213]) document state attr, add news file. refs [#5918](https://github.com/twisted/twisted/issues/5918)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

Added documentation and forced a [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/state-doc-5918). Up for review.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

Looks good. Please commit.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [37248]) Merge state-doc-5918: Documented the `state` attribute of StatefulStringProtocol.

Author: thijs
Reviewer: tom.prince
Fixes: [#5918](https://github.com/twisted/twisted/issues/5918)
